crates/arkworks_backend/src/bridge.rs

```diff
@@ -0,0 +1,150 @@
+use acvm::acir::{circuit::Circuit, native_types::Witness};
+use acvm::acir::{circuit::PublicInputs, native_types::Arithmetic};
+use ark_ff::Field;
+use ark_relations::{
+    lc,
+    r1cs::{
+        ConstraintSynthesizer, ConstraintSystemRef, LinearCombination, SynthesisError, Variable,
+    },
+};
+
+// AcirCircuit and AcirArithGate are structs that arkworks can synthesise.
+//
+// The difference between these structures and the ACIR structure that the compiler uses is the following:
+// - The compilers ACIR struct is currently fixed to bn254
+// - These structures only support arithmetic gates, while the compiler has other
+// gate types. These can be added later once the backend knows how to deal with things like XOR
+// or once ACIR is taught how to do convert these black box functions to Arithmetic gates.
+//
+// XXX: Ideally we want to implement `ConstraintSynthesizer` on ACIR however
+// this does not seem possible since ACIR is juts a description of the constraint system and the API Asks for prover values also.
+//
+// Perfect API would look like:
+// - index(srs, circ)
+// - prove(index_pk, prover_values, rng)
+// - verify(index_vk, verifier, rng)
+#[derive(Clone)]
+pub struct AcirCircuit<F: Field> {
```

Comment:
You don't have to make these generic over `F: Field`; you could just fill in `ark_bn254::Fr` here (and elsewhere in the file), no?

Reply:
In the end, the compiler will be generic over any Field, the only reason it is bn254 specific right now is because the default backend only supported this. I can fill in bn254 as you suggested and then in a different PR refactor for generics. This may be better as it is not clear when generic fields will land; 

- It first requires a PR that will allow backends to specify what Fields they can accept